### DISCUSSION TITLE
API for changing push policy ("Limit how many branches and tags can be updated in a single push")

### DISCUSSION THREAD

**Discussion by jakub-g** [community/community#47227] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hello, 

We've enabled the push policy in our repo to allow pushing max 5 refs at a time.

However, periodically we run a GH action where we'd like to delete a large number of tags (older than X weeks) at once.

So we'd like to be able to temporarily disable that push policy while in GH action; delete the tags; and reenable it.

Is there an API for that? I couldn't find one in the docs.

Thanks,
Jakub

**Comment by Sarvagya-24-chaturvedi** [comment]
Hi @jakub-g,

You don't need to disable or re-enable any repository policies for this!

The 5-ref limit only applies to **`git push`** commands over the Git wire protocol. It does **not** apply to the GitHub REST API.

Inside your GitHub Action, you can delete the tags directly using the built-in `gh` CLI:

```bash
# Delete a tag directly via API without triggering git push policies:
gh api --method DELETE /repos/${{ github.repository }}/git/refs/tags/<TAG_NAME>
